src/v/datalake/coordinator/coordinator.cc

```diff
@@ -847,4 +846,29 @@ coordinator::sync_get_usage_stats() {
     }
     co_return result;
 }
+
+ss::sstring coordinator::get_effective_default_partition_spec(
+  const std::optional<ss::sstring>& partition_spec) const {
+    const auto& cfg = config::shard_local_cfg();
+    auto current_spec = partition_spec.value_or(default_partition_spec_());
+
+    bool is_glue = cfg.iceberg_catalog_type()
+                     == config::datalake_catalog_type::rest
+                   && cfg.iceberg_rest_catalog_authentication_mode()
+                        == config::datalake_catalog_auth_mode::aws_sigv4
+                   && cfg.iceberg_rest_catalog_aws_service_name().value_or("")
+                        == "glue";
```

Comment:
In the [credential_manager](https://github.com/redpanda-data/redpanda/blob/8b6866b266962cd8668a7553591a430401d91b70/src/v/datalake/credential_manager.cc#L54-L55) it looks like we `value_or("glue")` to get the service name. Should we do the same here? Or should the property actually be non-optional?

Reply:
Makes sense, making it non-optional SGTM 👍